test/functional/editor/jump_spec.lua

```diff
@@ -182,6 +182,109 @@ describe("jumpoptions=stack behaves like 'tagstack'", function()
   end)
 end)
 
+describe('buffer deletion', function()
+  local base_file = 'Xtest-functional-buffer-deletion'
+  local file1 = base_file..'1'
+  local file2 = base_file..'2'
+  local file3 = base_file..'3'
+  local base_content = 'text'
+  local content1 = base_content..'1'
+  local content2 = base_content..'2'
+  local content3 = base_content..'3'
+  local function delete_buf(file)
+    local bufnr = funcs.bufnr(file)
+    command('bw '..bufnr)
+  end
+
+  before_each(function()
+    clear()
+    command('clearjumps')
+
+    write_file(file1, content1, false, false)
+    write_file(file2, content2, false, false)
+    write_file(file3, content3, false, false)
+
+    command("edit " .. file1)
+    command("edit " .. file2)
+    command("edit " .. file3)
+  end)
+
+  it('deletes jump list entries when the current buffer is deleted', function()
+    command("edit " .. file1)
+
+    eq(   ''
+       .. ' jump line  col file/text\n'
+       .. '   3     1    0 '..content1..'\n'
+       .. '   2     1    0 '..file2..'\n'
+       .. '   1     1    0 '..file3..'\n'
+       .. '>',
+       exec_capture('jumps'))
+
+    command('bw')
+
+    eq(   ''
+       .. ' jump line  col file/text\n'
+       .. '   1     1    0 '..file2..'\n'
+       .. '>  0     1    0 '..content3,
+       exec_capture('jumps'))
+  end)
+
+  it('deletes jump list entries when another buffer is deleted', function()
+    eq(   ''
+       .. ' jump line  col file/text\n'
+       .. '   2     1    0 '..file1..'\n'
+       .. '   1     1    0 '..file2..'\n'
+       .. '>',
```

Comment:
could use dedent() with a literaly `[[...]]` string.

Reply:
How would I do that nicely given that I have to format the string inside the `[[...]]`? I thought it would be nice not to have the file name/content hard coded in the string 30 times to make the code easier to maintain, but if that style of creating the literals is not in line with the rest of the tests I can change it.

Reply:
@justinmk I really don't want to be pushy and I know you're a busy guy but it would be really great if you could help me out here and tell me what else I need to do before merging

Reply:
> I thought it would be nice not to have the file name/content hard coded in the string

definitely. sometimes it's worth using a trick like replacing a token, e.g. `(([[foo {file1} ...]]):gsub('%{file1%}', 'x'))`

don't need to block on this, though.